
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1205200413 -7200
# Node ID 538f8892a2f159eb156808f037a0ed90e14ccd14
# Parent 975d78cadc1ed50adc9cd97b8b19faffb36b14fe
BODY/BODYSTRUCTURE fetch: Don't crash if we already had message parts
parsed.

--- a/src/lib-storage/index/index-mail.c	Sun Mar 09 12:50:15 2008 +0200
+++ b/src/lib-storage/index/index-mail.c	Tue Mar 11 03:53:33 2008 +0200
@@ -569,9 +569,8 @@ static void index_mail_parse_bodystructu
 			i_assert(data->parts->next == NULL);
 
 			old_offset = data->stream->v_offset;
-			i_stream_seek(data->stream,
-				      data->hdr_size.physical_size);
-			message_parse_from_parts(data->parts->children,
+			i_stream_seek(data->stream, 0);
+			message_parse_from_parts(data->parts,
 						data->stream,
 						parse_bodystructure_part_header,
 						mail->data_pool);

